### PR TITLE
Add required TLS for operator

### LINKED ISSUE #1278: Enable TLS client-server cert validation for k8s operator if mTLS is enabled
Today, the operator is running an HTTP service for the sidecars to connect to it and get components.

Although there is no sensitive information in the data returned from the operator, it would be good to reduce the attack surface area by establishing the connection over TLS with client-server verification.

This would prevent Man In The Middle attacks as well as rogue code inside the network attempting to fetch resources and bypass the Kubernetes RBAC mechanism.

Also, the operator should transition go gRPC instead of HTTP, and keep consistent with the rest of the Dapr system components.

### PR DESCRIPTION
**Pull request by yaron2**
Closes #1278 

This PR does the following:

1. Adds mandatory TLS between sidecar to Operator when mTLS is enabled
2. Updated tests
3. Removes some duplicate code
4. Filesystem watcher moved to separate package for reuse
5. Reuses operator gRPC connection for efficiency

### CODE REVIEW

File: pkg/runtime/cli.go [hunk 1]
```diff
@@ -120,10 +122,18 @@ func FromFlags() (*DaprRuntime, error) {
 	var globalConfig *global_config.Configuration
 	var configErr error
 
+	var certChain *certchain.CertChain
```
Comment (amanbha) [inline pkg/runtime/cli.go]: may be i am missing something here, but this variable certChain is not assigned any value, wouldn't it be nil always when passed to GetOperatorClient below(line 136) which would return a client without tls?
Should it pass runtimeConfig.CertChain instead as thats properly assigned.

Comment (yaron2) [reply pkg/runtime/cli.go]: When mTLS is disabled, it will return a non secured connection.

Comment (amanbha) [reply pkg/runtime/cli.go]: but this var "certChain" is not being set to anything when mTLS is enabled so GetOperatorClient  will always return a client without tls when this "certChain" var is passed to it.

Comment (yaron2) [reply pkg/runtime/cli.go]: oh good catch. fixed now.

**Pull request merged**